src/client/clientmap.cpp

```diff
@@ -494,15 +503,16 @@ void ClientMap::updateDrawList()
 				if (mesh_grid.cell_size > 1) {
 					// Block meshes are stored in the corner block of a chunk
 					// (where all coordinate are divisible by the chunk size)
-					// Add them to the de-dup set.
-					shortlist.emplace(mesh_grid.getMeshPos(block->getPos()));
-					// All other blocks we can grab and add to the keeplist right away.
-					m_keeplist.push_back(block);
-					block->refGrab();
+					// Deduplicate and add them later
+					shortlist.emplace(mesh_grid.getMeshPos(block_pos));
+					// All other blocks we can add to m_keeplist right away
+					if (!mesh_grid.isMeshPos(block_pos)) {
```

Comment:
Huh? We need to add all blocks to the keeplist. When the loops-culler is enabled touchMapBlocks does not do anything.

Reply:
I understood that the only purpose of the keeplist is to keep a reference for blocks.
Since mesh-having blocks end up in the drawlist (whose blocks are `refGrab`ed too), we don't need them in both, right?